src/sleep/sleep.c

```diff
@@ -390,7 +390,15 @@ static int custom_timer_suspend(const SleepConfig *sleep_config) {
                         }
                 }
 
-                /* Do not suspend more than HibernateDelaySec= */
+                /* Do not suspend more than HibernateDelaySec= unless HibernateOnACPower=no and currently on AC power */
+                if (!sleep_config->hibernate_on_ac_power) {
+                        /* Do not allow "decay" to suspend if the system has no battery. */
+                        if (hashmap_isempty(last_capacity))
+                                log_warning("HibernateOnACPower=no was ignored because the system does not have a battery.");
```

Comment:
Maybe use `log_once` here? We don't want multiple wakeups to spam the journal after all